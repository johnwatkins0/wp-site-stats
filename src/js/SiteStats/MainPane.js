/* eslint camelcase: 0 */
import React from 'react';
import format from 'date-fns/format';
import get from 'lodash.get';


export const getCommentExcerpt = ( comment ) => {
	const words = comment.split( ' ' );
	if ( 15 > words.length ) {
		return words.join( ' ' );
	}

	return `${words.slice( 0, 15 ).join( ' ' )} ...`;
};

export const Header = ({ site_title, site_url, activeSiteCreatedDate }) => (
	<header className="MainPane__header">
		<h2 dangerouslySetInnerHTML={ { __html: site_title || '&nbsp;' } } />
		<div className="MainPane__site-info">
			<div>
				<span className="MainPane__site-info-key">
					URL:
				</span>
				{' '}
				<a href={site_url}>
					{site_url}
				</a>
			</div>
			<div>
				<span className="MainPane__site-info-key">
					Created:
				</span>
				{' '}
				{format( activeSiteCreatedDate, 'MMMM D, YYYY' )}
			</div>
		</div>
	</header>
);

export const Stat = ({ title, number = '' }) => (
	<div className="MainPane__stat-container">
		<div className="MainPane__stat">
			<h4 className="MainPane__stat-title">
				{title}
			</h4>
			<div className="MainPane__stat-number">
				{number}
			</div>
		</div>
	</div>
);

export const Stats = ({ post_count, page_count, user_count, comment_count })  => (
	<section className="MainPane__section">
		<h3>By the Numbers</h3>
		<div  className="MainPane__stats">
			<Stat title="Posts" number={get( post_count, 'publish', '-' )} />
			<Stat title="Pages" number={get( page_count, 'publish', '-' )} />
			<Stat title="Users" number={user_count || '-'} />
			<Stat title="Comments" number={get( comment_count, 'approved', '-' )} />
		</div>
	</section>
);

export const LatestPost = ({ link, title = '' }) => (
	<section className="MainPane__section MainPane__latest-post">
		<h3>
			Latest Post
		</h3>
		<div className="MainPane__latest-post-link">
			<a
				href={link}
				dangerouslySetInnerHTML={
					{
						__html: title.rendered || '(no title)'
					}
				}
			/>
		</div>
	</section>
);

export const LatestComment = ({ link, author_name = '', date }) => (
	<section className="MainPane__section MainPane__latest-comment">
		<h3>
			Latest Comment
		</h3>
		<div className="MainPane__latest-comment-link">
			<a href={link}>
				{
					`${author_name} 
					on ${format( date, 'MMMM D, YYYY' )}
					at ${format( date, 'h:mm aa' )}`
				}
			</a>
		</div>
	</section>
);

export const Footer = ({ shouldRefresh, secondsToRefresh }) => {
	if ( ! shouldRefresh ) {
		return null;
	}

	return (
		<footer className="MainPane__footer">
			Refreshing in {secondsToRefresh} second{1 === secondsToRefresh ? '' : 's'}.
		</footer>
	);
};

const MainPane = ({
	site_title = '&nbsp;',
	site_url,
	user_count,
	post_count,
	page_count,
	latest_post,
	latest_comment,
	comment_count,
	activeSiteCreatedDate,
	secondsToRefresh,
	shouldRefresh
}) => (
	<article className="MainPane">
		<Header { ...{ site_title, site_url, activeSiteCreatedDate } } />
		<Stats { ...{ post_count, page_count, user_count, comment_count } } />
		<LatestPost { ...latest_post } />
		<LatestComment { ...latest_comment } />
		<Footer { ...{ shouldRefresh, secondsToRefresh } } />
	</article>
);

export default MainPane;
