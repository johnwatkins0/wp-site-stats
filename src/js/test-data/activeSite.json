{
    "site_title": "My Site",
    "site_url": "http://wpmu/my-site",
    "user_count": 1,
    "post_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "page_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "comment_count": {
        "approved": "1",
        "spam": 0,
        "trash": 0,
        "post-trashed": 0,
        "total_comments": 1,
        "all": 1,
        "moderated": 0
    },
    "latest_post": {
        "id": 1,
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "guid": {
            "rendered": "http://wpmu/my-site/?p=1"
        },
        "modified": "2018-04-02T14:29:11",
        "modified_gmt": "2018-04-02T14:29:11",
        "slug": "hello-world",
        "status": "publish",
        "type": "post",
        "link": "http://wpmu/my-site/2018/04/02/hello-world/",
        "title": {
            "rendered": "Hello world!"
        },
        "content": {
            "rendered": "<p>Welcome to <a href=\"http://wpmu/\">Colby College Sites</a>. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "excerpt": {
            "rendered": "<p>Welcome to Colby College Sites. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "author": 1,
        "featured_media": 0,
        "comment_status": "open",
        "ping_status": "open",
        "sticky": false,
        "template": "",
        "format": "standard",
        "meta": [],
        "categories": [
            1
        ],
        "tags": [],
        "revisions": null
    },
    "latest_comment": {
        "id": 1,
        "post": 1,
        "parent": 0,
        "author": 0,
        "author_name": "A WordPress Commenter",
        "author_url": "http://wpmu/",
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "content": {
            "rendered": "<p>Hi, this is a comment.<br />\nTo get started with moderating, editing, and deleting comments, please visit the Comments screen in the dashboard.<br />\nCommenter avatars come from <a href=\"https://gravatar.com\">Gravatar</a>.</p>\n"
        },
        "link": "http://wpmu/my-site/2018/04/02/hello-world/#comment-1",
        "status": "approved",
        "type": "comment",
        "author_avatar_urls": {
            "24": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=24&d=mm&r=g",
            "48": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=48&d=mm&r=g",
            "96": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=96&d=mm&r=g"
        },
        "meta": []
    }
}