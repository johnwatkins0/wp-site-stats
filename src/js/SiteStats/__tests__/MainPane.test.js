import React from 'react';
import renderer from 'react-test-renderer';

import MainPane from '../MainPane';

import activeSite from '../../test-data/activeSite.json';

test( 'MainPane component', () => {
	const tree = renderer.create(
		<MainPane
            {...activeSite}
		/>
	).toJSON();
	expect( tree ).toMatchSnapshot();
});
