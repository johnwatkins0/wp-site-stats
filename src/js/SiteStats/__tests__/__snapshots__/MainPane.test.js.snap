// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPane component renders correctly 1`] = `
<article
  className="MainPane"
>
  <header
    className="MainPane__header"
  >
    <h2
      dangerouslySetInnerHTML={
        Object {
          "__html": "&nbsp;",
        }
      }
    />
    <div
      className="MainPane__site-info"
    >
      <div>
        <span
          className="MainPane__site-info-key"
        >
          URL:
        </span>
         
        <a
          href={undefined}
        />
      </div>
    </div>
  </header>
  <section
    className="MainPane__section"
  >
    <h3>
      By the Numbers
    </h3>
    <div
      className="MainPane__stats"
    >
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Posts
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Pages
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Users
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Comments
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
    </div>
  </section>
</article>
`;

exports[`MainPane component renders correctly with no footer 1`] = `
<article
  className="MainPane"
>
  <header
    className="MainPane__header"
  >
    <h2
      dangerouslySetInnerHTML={
        Object {
          "__html": "&nbsp;",
        }
      }
    />
    <div
      className="MainPane__site-info"
    >
      <div>
        <span
          className="MainPane__site-info-key"
        >
          URL:
        </span>
         
        <a
          href={undefined}
        />
      </div>
    </div>
  </header>
  <section
    className="MainPane__section"
  >
    <h3>
      By the Numbers
    </h3>
    <div
      className="MainPane__stats"
    >
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Posts
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Pages
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Users
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
      <div
        className="MainPane__stat-container"
      >
        <div
          className="MainPane__stat"
        >
          <h4
            className="MainPane__stat-title"
          >
            Comments
          </h4>
          <div
            className="MainPane__stat-number"
          >
            -
          </div>
        </div>
      </div>
    </div>
  </section>
</article>
`;
